src/nvim/api/extmark.h

```diff
@@ -4,14 +4,26 @@
 
 #include "nvim/api/keysets_defs.h"  // IWYU pragma: keep
 #include "nvim/api/private/defs.h"  // IWYU pragma: keep
+#include "nvim/buffer_defs.h"
 #include "nvim/decoration_defs.h"  // IWYU pragma: keep
 #include "nvim/macros_defs.h"
 #include "nvim/map_defs.h"
 #include "nvim/types_defs.h"
 
 EXTERN Map(String, int) namespace_ids INIT( = MAP_INIT);
+EXTERN Set(uint32_t) namespace_scoped INIT( = SET_INIT);
```

Comment:
Is it too early to introduce (pseudocode):

    namespaces: Map<id, NamespaceProps>

where `NamespaceProps` is a struct like:

```
{ scope: NsScope; ... }
```